packages/embeds/embed-core/embed-iframe-init.ts

```diff
@@ -1,34 +1,38 @@
+import type { EmbedThemeConfig } from "src/embed-iframe";
+
 export default function EmbedInitIframe() {
   if (typeof window === "undefined" || window.isEmbed) {
     return;
   }
 
-  const calEmbedMode = typeof new URL(document.URL).searchParams.get("embed") === "string";
+  const embedNameSpaceFromQueryParam = new URL(document.URL).searchParams.get("embed");
 
-  /* Iframe Name */
-  window.name.includes("cal-embed");
+  // Namespace is initially set in query param `embed` but the query param might get lost during soft navigation
+  // So, we also check for the namespace in `window.name` which is set when iframe is created by embed.ts and persists for the duration of iframe's life
+  // Note that, window.name isn't lost during hard navigation as well. Though, hard navigation isn't something that would happen in the app, but it's critical to be able to detect embed mode even after that(just in case)
+  // We might just use window.name but if just in case something resets the `window.name`, we will still have the namespace in query param
+  const embedNamespace =
+    typeof embedNameSpaceFromQueryParam === "string"
+      ? embedNameSpaceFromQueryParam
+      : window.name.replace(/cal-embed=(.*)/, "$1").trim();
```

Comment:
It is set here https://github.com/calcom/cal.com/pull/9085/files#diff-e895c26df52266ecfd6334060b1dc6fa05cac4106eee7f713d0a07472cc08d07R218